lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -100,15 +96,12 @@ def attributes_for_serializer(serializer, options)
 
         def resource_object_for(serializer, options)
           options[:fields] = @fieldset && @fieldset.fields_for(serializer)
-          options[:required_fields] = [:id, :json_api_type]
 
           cache_check(serializer) do
             attributes = serializer.attributes(options)
+            attributes.delete(:id)
 
-            result = {
-              id: attributes.delete(:id).to_s,
-              type: attributes.delete(:json_api_type)
-            }
+            result = resource_identifier(serializer)
```

Comment:
No, I get that.. but you're not using it, but it's there... that's what I mean by throwing it out... I'm sure there's a detail I'm missing in the refactor, but figured it'd be good to ask from ignorance :)